{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "legal-lending",
   "metadata": {},
   "source": [
    "Can only do this analysis, involving colinearity, for genomes with genome mapped to chromosomes."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "sound-cooperation",
   "metadata": {},
   "source": [
    "Might not be that useful to do it on more species. Antynign not in briggsae clade will be veery distantly related. Fewer orthologs --> less power. Also very likely that rearrangements will occur more than once. Multiple rearrangements of the same type. "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "amber-webmaster",
   "metadata": {},
   "source": [
    "ftp://ftp.wormbase.org/pub/wormbase/releases/WS275/species/c_inopinata/PRJDB5687/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "tired-variable",
   "metadata": {},
   "outputs": [],
   "source": [
    "worm='inopinata'\n",
    "ts_gtf='~/part_ii_project/canonical_genesets/c_inopinata.PRJDB5687.WS275.canonical_geneset.gtf'\n",
    "worm_fasta_file='c_inopinata.PRJDB5687.WS275.CDS_transcripts.fa'\n",
    "protein_fasta_file='c_inopinata.PRJDB5687.WS275.protein.fa'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "earlier-swaziland",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "from Bio import SeqIO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "moderate-oakland",
   "metadata": {},
   "outputs": [],
   "source": [
    "chrom_bed_columns=['chrom','chromStart','chromEnd','name',\n",
    "       'score','strand','thickStart','thickEnd','itemRgb']\n",
    "bed_12_columns=['chrom','chromStart','chromEnd','name',\n",
    "       'score','strand','thickStart','thickEnd','itemRgb',\n",
    "        'blockCount', 'blockSizes', 'blockStart']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "beautiful-instrumentation",
   "metadata": {},
   "source": [
    "# Make transcript to gene map for the worm"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "numerous-rogers",
   "metadata": {},
   "source": [
    "Will have to figure this out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "nearby-stress",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(worm_fasta_file) as fasta_file:\n",
    "    descriptions=[]\n",
    "    for seq_record in SeqIO.parse(fasta_file, 'fasta'):\n",
    "        descriptions.append(seq_record.description)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "experimental-mapping",
   "metadata": {},
   "outputs": [],
   "source": [
    "transcripts=[]\n",
    "genes=[]\n",
    "\n",
    "for i in descriptions:\n",
    "    transcripts.append(i.split(' gene=')[0])\n",
    "    genes.append(i.split(' gene=')[1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "infrared-crystal",
   "metadata": {},
   "outputs": [],
   "source": [
    "ts_gs=pd.DataFrame({'genes':genes, 'transcripts':transcripts})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "embedded-shield",
   "metadata": {},
   "outputs": [],
   "source": [
    "ts_gs.to_csv('~/part_ii_project/canonical_genesets/'+worm+'_transcripts_and_genes.txt', \n",
    "             header=None,index=None, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "peripheral-alliance",
   "metadata": {},
   "source": [
    "# Make genes file for the worm"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bizarre-fishing",
   "metadata": {},
   "source": [
    "### Make a bed file of transcripts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "sophisticated-virgin",
   "metadata": {},
   "outputs": [],
   "source": [
    "# convert gtf to genepred to bed\n",
    "\n",
    "gene_pred_file='~/part_ii_project/canonical_genesets/'+worm+'_genePred'\n",
    "\n",
    "! ./gtfToGenePred {ts_gtf} {gene_pred_file}\n",
    "\n",
    "ts_bed='~/part_ii_project/canonical_genesets/'+worm+'_transcripts.bed'\n",
    "\n",
    "! ./genePredToBed {gene_pred_file} {ts_bed}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "genuine-sitting",
   "metadata": {},
   "outputs": [],
   "source": [
    "ts=pd.read_csv(ts_bed,sep='\\t',names=bed_12_columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "acoustic-vertex",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>chromStart</th>\n",
       "      <th>chromEnd</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>thickStart</th>\n",
       "      <th>thickEnd</th>\n",
       "      <th>itemRgb</th>\n",
       "      <th>blockCount</th>\n",
       "      <th>blockSizes</th>\n",
       "      <th>blockStart</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Sp34_Chr5</td>\n",
       "      <td>5859</td>\n",
       "      <td>7266</td>\n",
       "      <td>Sp34_50000100.t1</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>5859</td>\n",
       "      <td>7266</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>66,183,</td>\n",
       "      <td>0,1224,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Sp34_Chr5</td>\n",
       "      <td>9326</td>\n",
       "      <td>10190</td>\n",
       "      <td>Sp34_50000200.t1</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "      <td>9326</td>\n",
       "      <td>10190</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>63,56,97,</td>\n",
       "      <td>0,654,767,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Sp34_Chr5</td>\n",
       "      <td>11792</td>\n",
       "      <td>12182</td>\n",
       "      <td>Sp34_50000300.t1</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "      <td>11792</td>\n",
       "      <td>12182</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>390,</td>\n",
       "      <td>0,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Sp34_Chr5</td>\n",
       "      <td>15692</td>\n",
       "      <td>16990</td>\n",
       "      <td>Sp34_50000400.t1</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>15692</td>\n",
       "      <td>16990</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>193,423,236,88,50,</td>\n",
       "      <td>0,378,841,1119,1248,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Sp34_Chr5</td>\n",
       "      <td>21042</td>\n",
       "      <td>23817</td>\n",
       "      <td>Sp34_50000500.t1</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>21042</td>\n",
       "      <td>23817</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>16,63,66,117,56,</td>\n",
       "      <td>0,434,1323,2142,2719,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21604</th>\n",
       "      <td>Sp34_ChrX</td>\n",
       "      <td>18172142</td>\n",
       "      <td>18173045</td>\n",
       "      <td>Sp34_X0278900.t1</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "      <td>18172142</td>\n",
       "      <td>18173045</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>903,</td>\n",
       "      <td>0,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21605</th>\n",
       "      <td>Sp34_ChrX</td>\n",
       "      <td>18173649</td>\n",
       "      <td>18175323</td>\n",
       "      <td>Sp34_X0279000.t1</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "      <td>18173649</td>\n",
       "      <td>18175323</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1674,</td>\n",
       "      <td>0,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21606</th>\n",
       "      <td>Sp34_ChrX</td>\n",
       "      <td>18176970</td>\n",
       "      <td>18177814</td>\n",
       "      <td>Sp34_X0279100.t1</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "      <td>18176970</td>\n",
       "      <td>18177814</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>21,59,95,47,82,116,</td>\n",
       "      <td>0,89,215,420,569,728,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21607</th>\n",
       "      <td>Sp34_ChrX</td>\n",
       "      <td>18178768</td>\n",
       "      <td>18183524</td>\n",
       "      <td>Sp34_X0279200.t1</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>18178768</td>\n",
       "      <td>18183524</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>2040,540,270,717,</td>\n",
       "      <td>0,2099,2763,4039,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21608</th>\n",
       "      <td>Sp34_ChrX</td>\n",
       "      <td>18183562</td>\n",
       "      <td>18187184</td>\n",
       "      <td>Sp34_X0279300.t1</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>18183562</td>\n",
       "      <td>18187184</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>237,352,184,775,</td>\n",
       "      <td>0,1950,2348,2847,</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>21609 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           chrom  chromStart  chromEnd              name  score strand  \\\n",
       "0      Sp34_Chr5        5859      7266  Sp34_50000100.t1      0      -   \n",
       "1      Sp34_Chr5        9326     10190  Sp34_50000200.t1      0      +   \n",
       "2      Sp34_Chr5       11792     12182  Sp34_50000300.t1      0      +   \n",
       "3      Sp34_Chr5       15692     16990  Sp34_50000400.t1      0      -   \n",
       "4      Sp34_Chr5       21042     23817  Sp34_50000500.t1      0      -   \n",
       "...          ...         ...       ...               ...    ...    ...   \n",
       "21604  Sp34_ChrX    18172142  18173045  Sp34_X0278900.t1      0      +   \n",
       "21605  Sp34_ChrX    18173649  18175323  Sp34_X0279000.t1      0      +   \n",
       "21606  Sp34_ChrX    18176970  18177814  Sp34_X0279100.t1      0      +   \n",
       "21607  Sp34_ChrX    18178768  18183524  Sp34_X0279200.t1      0      -   \n",
       "21608  Sp34_ChrX    18183562  18187184  Sp34_X0279300.t1      0      -   \n",
       "\n",
       "       thickStart  thickEnd  itemRgb  blockCount           blockSizes  \\\n",
       "0            5859      7266        0           2              66,183,   \n",
       "1            9326     10190        0           3            63,56,97,   \n",
       "2           11792     12182        0           1                 390,   \n",
       "3           15692     16990        0           5   193,423,236,88,50,   \n",
       "4           21042     23817        0           5     16,63,66,117,56,   \n",
       "...           ...       ...      ...         ...                  ...   \n",
       "21604    18172142  18173045        0           1                 903,   \n",
       "21605    18173649  18175323        0           1                1674,   \n",
       "21606    18176970  18177814        0           6  21,59,95,47,82,116,   \n",
       "21607    18178768  18183524        0           4    2040,540,270,717,   \n",
       "21608    18183562  18187184        0           4     237,352,184,775,   \n",
       "\n",
       "                  blockStart  \n",
       "0                    0,1224,  \n",
       "1                 0,654,767,  \n",
       "2                         0,  \n",
       "3       0,378,841,1119,1248,  \n",
       "4      0,434,1323,2142,2719,  \n",
       "...                      ...  \n",
       "21604                     0,  \n",
       "21605                     0,  \n",
       "21606  0,89,215,420,569,728,  \n",
       "21607      0,2099,2763,4039,  \n",
       "21608      0,1950,2348,2847,  \n",
       "\n",
       "[21609 rows x 12 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ts"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "serial-handbook",
   "metadata": {},
   "source": [
    "### Rerformat chromosome names if necessary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "skilled-fusion",
   "metadata": {},
   "outputs": [],
   "source": [
    "chroms=[i.split('_Chr')[1] for i in list(ts['chrom'])]\n",
    "ts['chrom']=chroms\n",
    "\n",
    "arabic_to_rom={'1':'I', '2':'II', '3':'III', '4':'IV', '5':'V', 'X':'X'}\n",
    "chroms=[arabic_to_rom[i] for i in list(ts['chrom'])]\n",
    "ts['chrom']=chroms"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "occasional-crown",
   "metadata": {},
   "source": [
    "### Add gene names instead of transcript names and have just 1 row per gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "facial-effort",
   "metadata": {},
   "outputs": [],
   "source": [
    "# get a list of each gene just once\n",
    "\n",
    "one_of_each_gene=list(set(ts_gs['genes']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "mediterranean-transfer",
   "metadata": {},
   "outputs": [],
   "source": [
    "# add the gene IDs to the transcript data frame\n",
    "\n",
    "total_df=ts.copy()\n",
    "total_df['gene']=''\n",
    "\n",
    "for i in range(len(ts_gs)):\n",
    "    row=total_df.index[total_df['name']==ts_gs['transcripts'][i]]\n",
    "    total_df.at[row,'gene']=ts_gs['genes'][i]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "sporting-ratio",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>chromStart</th>\n",
       "      <th>chromEnd</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>thickStart</th>\n",
       "      <th>thickEnd</th>\n",
       "      <th>itemRgb</th>\n",
       "      <th>blockCount</th>\n",
       "      <th>blockSizes</th>\n",
       "      <th>blockStart</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>V</td>\n",
       "      <td>5859</td>\n",
       "      <td>7266</td>\n",
       "      <td>Sp34_50000100.t1</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>5859</td>\n",
       "      <td>7266</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>66,183,</td>\n",
       "      <td>0,1224,</td>\n",
       "      <td>Sp34_50000100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>V</td>\n",
       "      <td>9326</td>\n",
       "      <td>10190</td>\n",
       "      <td>Sp34_50000200.t1</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "      <td>9326</td>\n",
       "      <td>10190</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>63,56,97,</td>\n",
       "      <td>0,654,767,</td>\n",
       "      <td>Sp34_50000200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>V</td>\n",
       "      <td>11792</td>\n",
       "      <td>12182</td>\n",
       "      <td>Sp34_50000300.t1</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "      <td>11792</td>\n",
       "      <td>12182</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>390,</td>\n",
       "      <td>0,</td>\n",
       "      <td>Sp34_50000300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>V</td>\n",
       "      <td>15692</td>\n",
       "      <td>16990</td>\n",
       "      <td>Sp34_50000400.t1</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>15692</td>\n",
       "      <td>16990</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>193,423,236,88,50,</td>\n",
       "      <td>0,378,841,1119,1248,</td>\n",
       "      <td>Sp34_50000400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>V</td>\n",
       "      <td>21042</td>\n",
       "      <td>23817</td>\n",
       "      <td>Sp34_50000500.t1</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>21042</td>\n",
       "      <td>23817</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>16,63,66,117,56,</td>\n",
       "      <td>0,434,1323,2142,2719,</td>\n",
       "      <td>Sp34_50000500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21604</th>\n",
       "      <td>X</td>\n",
       "      <td>18172142</td>\n",
       "      <td>18173045</td>\n",
       "      <td>Sp34_X0278900.t1</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "      <td>18172142</td>\n",
       "      <td>18173045</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>903,</td>\n",
       "      <td>0,</td>\n",
       "      <td>Sp34_X0278900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21605</th>\n",
       "      <td>X</td>\n",
       "      <td>18173649</td>\n",
       "      <td>18175323</td>\n",
       "      <td>Sp34_X0279000.t1</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "      <td>18173649</td>\n",
       "      <td>18175323</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1674,</td>\n",
       "      <td>0,</td>\n",
       "      <td>Sp34_X0279000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21606</th>\n",
       "      <td>X</td>\n",
       "      <td>18176970</td>\n",
       "      <td>18177814</td>\n",
       "      <td>Sp34_X0279100.t1</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "      <td>18176970</td>\n",
       "      <td>18177814</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>21,59,95,47,82,116,</td>\n",
       "      <td>0,89,215,420,569,728,</td>\n",
       "      <td>Sp34_X0279100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21607</th>\n",
       "      <td>X</td>\n",
       "      <td>18178768</td>\n",
       "      <td>18183524</td>\n",
       "      <td>Sp34_X0279200.t1</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>18178768</td>\n",
       "      <td>18183524</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>2040,540,270,717,</td>\n",
       "      <td>0,2099,2763,4039,</td>\n",
       "      <td>Sp34_X0279200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21608</th>\n",
       "      <td>X</td>\n",
       "      <td>18183562</td>\n",
       "      <td>18187184</td>\n",
       "      <td>Sp34_X0279300.t1</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>18183562</td>\n",
       "      <td>18187184</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>237,352,184,775,</td>\n",
       "      <td>0,1950,2348,2847,</td>\n",
       "      <td>Sp34_X0279300</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>21609 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      chrom  chromStart  chromEnd              name  score strand  thickStart  \\\n",
       "0         V        5859      7266  Sp34_50000100.t1      0      -        5859   \n",
       "1         V        9326     10190  Sp34_50000200.t1      0      +        9326   \n",
       "2         V       11792     12182  Sp34_50000300.t1      0      +       11792   \n",
       "3         V       15692     16990  Sp34_50000400.t1      0      -       15692   \n",
       "4         V       21042     23817  Sp34_50000500.t1      0      -       21042   \n",
       "...     ...         ...       ...               ...    ...    ...         ...   \n",
       "21604     X    18172142  18173045  Sp34_X0278900.t1      0      +    18172142   \n",
       "21605     X    18173649  18175323  Sp34_X0279000.t1      0      +    18173649   \n",
       "21606     X    18176970  18177814  Sp34_X0279100.t1      0      +    18176970   \n",
       "21607     X    18178768  18183524  Sp34_X0279200.t1      0      -    18178768   \n",
       "21608     X    18183562  18187184  Sp34_X0279300.t1      0      -    18183562   \n",
       "\n",
       "       thickEnd  itemRgb  blockCount           blockSizes  \\\n",
       "0          7266        0           2              66,183,   \n",
       "1         10190        0           3            63,56,97,   \n",
       "2         12182        0           1                 390,   \n",
       "3         16990        0           5   193,423,236,88,50,   \n",
       "4         23817        0           5     16,63,66,117,56,   \n",
       "...         ...      ...         ...                  ...   \n",
       "21604  18173045        0           1                 903,   \n",
       "21605  18175323        0           1                1674,   \n",
       "21606  18177814        0           6  21,59,95,47,82,116,   \n",
       "21607  18183524        0           4    2040,540,270,717,   \n",
       "21608  18187184        0           4     237,352,184,775,   \n",
       "\n",
       "                  blockStart           gene  \n",
       "0                    0,1224,  Sp34_50000100  \n",
       "1                 0,654,767,  Sp34_50000200  \n",
       "2                         0,  Sp34_50000300  \n",
       "3       0,378,841,1119,1248,  Sp34_50000400  \n",
       "4      0,434,1323,2142,2719,  Sp34_50000500  \n",
       "...                      ...            ...  \n",
       "21604                     0,  Sp34_X0278900  \n",
       "21605                     0,  Sp34_X0279000  \n",
       "21606  0,89,215,420,569,728,  Sp34_X0279100  \n",
       "21607      0,2099,2763,4039,  Sp34_X0279200  \n",
       "21608      0,1950,2348,2847,  Sp34_X0279300  \n",
       "\n",
       "[21609 rows x 13 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "total_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "advanced-empty",
   "metadata": {},
   "outputs": [],
   "source": [
    "# These are the coding transcripts that we'll be looking at\n",
    "genes_df=total_df[total_df['gene']!=''].copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "controlled-belief",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get isoform sizes\n",
    "genes_df['transcript_size']=list(genes_df['chromEnd']-genes_df['chromStart'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "written-ratio",
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-19-03372e8eb0c7>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0mone_to_one\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mDataFrame\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mgenes_df\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mone_of_each_gene\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 5\u001b[0;31m     \u001b[0mto_compare\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mgenes_df\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mgenes_df\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'gene'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m==\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      6\u001b[0m     \u001b[0mlongest\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mto_compare\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'transcript_size'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0margmax\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      7\u001b[0m     \u001b[0mrow\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mto_compare\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0miloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mlongest\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/part_ii_project/lib/python3.7/site-packages/pandas/core/ops/common.py\u001b[0m in \u001b[0;36mnew_method\u001b[0;34m(self, other)\u001b[0m\n\u001b[1;32m     63\u001b[0m         \u001b[0mother\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mitem_from_zerodim\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mother\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     64\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 65\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mmethod\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mother\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     66\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     67\u001b[0m     \u001b[0;32mreturn\u001b[0m \u001b[0mnew_method\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/part_ii_project/lib/python3.7/site-packages/pandas/core/arraylike.py\u001b[0m in \u001b[0;36m__eq__\u001b[0;34m(self, other)\u001b[0m\n\u001b[1;32m     27\u001b[0m     \u001b[0;34m@\u001b[0m\u001b[0munpack_zerodim_and_defer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"__eq__\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     28\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__eq__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mother\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 29\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_cmp_method\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mother\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0moperator\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0meq\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     30\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     31\u001b[0m     \u001b[0;34m@\u001b[0m\u001b[0munpack_zerodim_and_defer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"__ne__\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/part_ii_project/lib/python3.7/site-packages/pandas/core/series.py\u001b[0m in \u001b[0;36m_cmp_method\u001b[0;34m(self, other, op)\u001b[0m\n\u001b[1;32m   4937\u001b[0m         \u001b[0mrvalues\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mextract_array\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mother\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mextract_numpy\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4938\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 4939\u001b[0;31m         \u001b[0mres_values\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mops\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcomparison_op\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlvalues\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrvalues\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mop\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   4940\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4941\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_construct_result\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mres_values\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mname\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mres_name\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/part_ii_project/lib/python3.7/site-packages/pandas/core/ops/array_ops.py\u001b[0m in \u001b[0;36mcomparison_op\u001b[0;34m(left, right, op)\u001b[0m\n\u001b[1;32m    229\u001b[0m         \u001b[0mres_values\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mop\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlvalues\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrvalues\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    230\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 231\u001b[0;31m     \u001b[0;32melif\u001b[0m \u001b[0mis_scalar\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrvalues\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0misna\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrvalues\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    232\u001b[0m         \u001b[0;31m# numpy does not like comparisons vs None\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    233\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mop\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0moperator\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mne\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/part_ii_project/lib/python3.7/site-packages/pandas/core/dtypes/missing.py\u001b[0m in \u001b[0;36misna\u001b[0;34m(obj)\u001b[0m\n\u001b[1;32m    125\u001b[0m     \u001b[0mName\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mbool\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    126\u001b[0m     \"\"\"\n\u001b[0;32m--> 127\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0m_isna\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    128\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    129\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/part_ii_project/lib/python3.7/site-packages/pandas/core/dtypes/missing.py\u001b[0m in \u001b[0;36m_isna\u001b[0;34m(obj, inf_as_na)\u001b[0m\n\u001b[1;32m    151\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mlibmissing\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mchecknull_old\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    152\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 153\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mlibmissing\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mchecknull\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    154\u001b[0m     \u001b[0;31m# hack (for now) because MI registers as ndarray\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    155\u001b[0m     \u001b[0;32melif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mABCMultiIndex\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "# make a table of just the longest isoforms of each gene\n",
    "\n",
    "one_to_one=pd.DataFrame(columns=genes_df.columns)\n",
    "for i in one_of_each_gene:\n",
    "    to_compare=genes_df[genes_df['gene']==i]\n",
    "    longest=to_compare['transcript_size'].argmax()\n",
    "    row=to_compare.iloc[longest,:]\n",
    "    to_concat=row.to_frame().T\n",
    "    one_to_one=pd.concat([one_to_one, to_concat])\n",
    "    \n",
    "path='~/part_ii_project/canonical_genesets/'+worm+'_one_to_one_with_transcript_name.bed'\n",
    "one_to_one.to_csv(path, sep='\\t', index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "metric-adoption",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>chromStart</th>\n",
       "      <th>chromEnd</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>thickStart</th>\n",
       "      <th>thickEnd</th>\n",
       "      <th>itemRgb</th>\n",
       "      <th>blockCount</th>\n",
       "      <th>blockSizes</th>\n",
       "      <th>blockStart</th>\n",
       "      <th>gene</th>\n",
       "      <th>transcript_size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>17709</th>\n",
       "      <td>III</td>\n",
       "      <td>13691959</td>\n",
       "      <td>13693971</td>\n",
       "      <td>Sp34_30238000.t1</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "      <td>13691959</td>\n",
       "      <td>13693971</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>92,51,112,158,79,150,132,</td>\n",
       "      <td>0,221,323,711,914,1050,1880,</td>\n",
       "      <td>Sp34_30238000</td>\n",
       "      <td>2012</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17422</th>\n",
       "      <td>III</td>\n",
       "      <td>12066194</td>\n",
       "      <td>12068990</td>\n",
       "      <td>Sp34_30210020.t1</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>12066194</td>\n",
       "      <td>12068990</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>425,124,226,410,22,35,</td>\n",
       "      <td>0,481,650,922,2374,2761,</td>\n",
       "      <td>Sp34_30210020</td>\n",
       "      <td>2796</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3217</th>\n",
       "      <td>V</td>\n",
       "      <td>16455650</td>\n",
       "      <td>16458211</td>\n",
       "      <td>Sp34_50307600.t1</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>16455650</td>\n",
       "      <td>16458211</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>268,82,92,121,67,</td>\n",
       "      <td>0,328,465,1188,2494,</td>\n",
       "      <td>Sp34_50307600</td>\n",
       "      <td>2561</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20864</th>\n",
       "      <td>X</td>\n",
       "      <td>13659551</td>\n",
       "      <td>13664714</td>\n",
       "      <td>Sp34_X0207000.t1</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>13659551</td>\n",
       "      <td>13664714</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>167,78,136,84,83,89,117,105,91,106,</td>\n",
       "      <td>0,1900,2025,2207,2695,2837,3042,3819,3979,5057,</td>\n",
       "      <td>Sp34_X0207000</td>\n",
       "      <td>5163</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17706</th>\n",
       "      <td>III</td>\n",
       "      <td>13680531</td>\n",
       "      <td>13683192</td>\n",
       "      <td>Sp34_30237700.t1</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>13680531</td>\n",
       "      <td>13683192</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2661,</td>\n",
       "      <td>0,</td>\n",
       "      <td>Sp34_30237700</td>\n",
       "      <td>2661</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4831</th>\n",
       "      <td>IV</td>\n",
       "      <td>1684102</td>\n",
       "      <td>1686767</td>\n",
       "      <td>Sp34_40031200.t1</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "      <td>1684102</td>\n",
       "      <td>1686767</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>25,110,157,368,299,666,145,</td>\n",
       "      <td>0,616,781,987,1405,1805,2520,</td>\n",
       "      <td>Sp34_40031200</td>\n",
       "      <td>2665</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5862</th>\n",
       "      <td>IV</td>\n",
       "      <td>8344899</td>\n",
       "      <td>8345855</td>\n",
       "      <td>Sp34_40131210.t1</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>8344899</td>\n",
       "      <td>8345855</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>157,196,490,</td>\n",
       "      <td>0,217,466,</td>\n",
       "      <td>Sp34_40131210</td>\n",
       "      <td>956</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15672</th>\n",
       "      <td>III</td>\n",
       "      <td>2682125</td>\n",
       "      <td>2682461</td>\n",
       "      <td>Sp34_30039900.t1</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>2682125</td>\n",
       "      <td>2682461</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>336,</td>\n",
       "      <td>0,</td>\n",
       "      <td>Sp34_30039900</td>\n",
       "      <td>336</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1863</th>\n",
       "      <td>V</td>\n",
       "      <td>9775688</td>\n",
       "      <td>9776590</td>\n",
       "      <td>Sp34_50177500.t1</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>9775688</td>\n",
       "      <td>9776590</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>113,94,18,</td>\n",
       "      <td>0,213,884,</td>\n",
       "      <td>Sp34_50177500</td>\n",
       "      <td>902</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15288</th>\n",
       "      <td>III</td>\n",
       "      <td>223446</td>\n",
       "      <td>225250</td>\n",
       "      <td>Sp34_30002600.t1</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "      <td>223446</td>\n",
       "      <td>225250</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>166,519,474,194,</td>\n",
       "      <td>0,487,1050,1610,</td>\n",
       "      <td>Sp34_30002600</td>\n",
       "      <td>1804</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>21443 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      chrom chromStart  chromEnd              name score strand thickStart  \\\n",
       "17709   III   13691959  13693971  Sp34_30238000.t1     0      +   13691959   \n",
       "17422   III   12066194  12068990  Sp34_30210020.t1     0      -   12066194   \n",
       "3217      V   16455650  16458211  Sp34_50307600.t1     0      -   16455650   \n",
       "20864     X   13659551  13664714  Sp34_X0207000.t1     0      -   13659551   \n",
       "17706   III   13680531  13683192  Sp34_30237700.t1     0      -   13680531   \n",
       "...     ...        ...       ...               ...   ...    ...        ...   \n",
       "4831     IV    1684102   1686767  Sp34_40031200.t1     0      +    1684102   \n",
       "5862     IV    8344899   8345855  Sp34_40131210.t1     0      -    8344899   \n",
       "15672   III    2682125   2682461  Sp34_30039900.t1     0      -    2682125   \n",
       "1863      V    9775688   9776590  Sp34_50177500.t1     0      -    9775688   \n",
       "15288   III     223446    225250  Sp34_30002600.t1     0      +     223446   \n",
       "\n",
       "       thickEnd itemRgb blockCount                           blockSizes  \\\n",
       "17709  13693971       0          7            92,51,112,158,79,150,132,   \n",
       "17422  12068990       0          6               425,124,226,410,22,35,   \n",
       "3217   16458211       0          5                    268,82,92,121,67,   \n",
       "20864  13664714       0         10  167,78,136,84,83,89,117,105,91,106,   \n",
       "17706  13683192       0          1                                2661,   \n",
       "...         ...     ...        ...                                  ...   \n",
       "4831    1686767       0          7          25,110,157,368,299,666,145,   \n",
       "5862    8345855       0          3                         157,196,490,   \n",
       "15672   2682461       0          1                                 336,   \n",
       "1863    9776590       0          3                           113,94,18,   \n",
       "15288    225250       0          4                     166,519,474,194,   \n",
       "\n",
       "                                            blockStart           gene  \\\n",
       "17709                     0,221,323,711,914,1050,1880,  Sp34_30238000   \n",
       "17422                         0,481,650,922,2374,2761,  Sp34_30210020   \n",
       "3217                              0,328,465,1188,2494,  Sp34_50307600   \n",
       "20864  0,1900,2025,2207,2695,2837,3042,3819,3979,5057,  Sp34_X0207000   \n",
       "17706                                               0,  Sp34_30237700   \n",
       "...                                                ...            ...   \n",
       "4831                     0,616,781,987,1405,1805,2520,  Sp34_40031200   \n",
       "5862                                        0,217,466,  Sp34_40131210   \n",
       "15672                                               0,  Sp34_30039900   \n",
       "1863                                        0,213,884,  Sp34_50177500   \n",
       "15288                                 0,487,1050,1610,  Sp34_30002600   \n",
       "\n",
       "      transcript_size  \n",
       "17709            2012  \n",
       "17422            2796  \n",
       "3217             2561  \n",
       "20864            5163  \n",
       "17706            2661  \n",
       "...               ...  \n",
       "4831             2665  \n",
       "5862              956  \n",
       "15672             336  \n",
       "1863              902  \n",
       "15288            1804  \n",
       "\n",
       "[21443 rows x 14 columns]"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "one_to_one"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "warming-speech",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_bed=one_to_one.copy().reset_index(drop=True)\n",
    "gene_bed['name']=gene_bed['gene']\n",
    "gene_bed.drop('transcript_size', axis=1).drop('gene', axis=1)\n",
    "\n",
    "path='~/part_ii_project/canonical_genesets/'+worm+'_one_of_each_gene.bed'\n",
    "gene_bed.to_csv(path, sep='\\t', index=None, header=None)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "brave-underground",
   "metadata": {},
   "source": [
    "# Make files of orthologous genes with and without operons"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "patent-seller",
   "metadata": {},
   "outputs": [],
   "source": [
    "bed_12_columns=['chrom','chromStart','chromEnd','name',\n",
    "       'score','strand','thickStart','thickEnd','itemRgb',\n",
    "        'blockCount', 'blockSizes', 'blockStart']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "reasonable-wildlife",
   "metadata": {},
   "outputs": [],
   "source": [
    "worm_genes=pd.read_csv('~/part_ii_project/canonical_genesets/'+worm+'_one_of_each_gene.bed',\n",
    "                           names=bed_12_columns, sep='\\t')\n",
    "elegans_genes=pd.read_csv('~/part_ii_project/canonical_genesets/elegans_one_of_each_gene.bed',\n",
    "                           names=bed_12_columns, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "iraqi-bishop",
   "metadata": {},
   "outputs": [],
   "source": [
    "operon_names=['gene_name','operon','blockCount','blockSizes','blockStart',\n",
    "              'chrom','gene_end','gene_start','itemRgb','score','strand','thickEnd',\n",
    "              'thickStart','position_in_operon']\n",
    "\n",
    "operons=pd.read_csv('~/part_ii_project/operon_information/every_gene_and_its_operon',names=operon_names,sep='\\t')\n",
    "\n",
    "indices=[]\n",
    "for index, row in elegans_genes.iterrows():\n",
    "    if row['name'] not in list(operons['gene_name']):indices.append(index)\n",
    "        \n",
    "elegans_genes_no_ops=elegans_genes.iloc[indices,:].copy().reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "sunset-mainstream",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(16430, 20040)"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(elegans_genes_no_ops), len(elegans_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "double-mobile",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>elegans</th>\n",
       "      <th>inopinata</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>WBGene00000001</td>\n",
       "      <td>Sp34_10257700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>WBGene00000002</td>\n",
       "      <td>Sp34_40340700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>WBGene00000003</td>\n",
       "      <td>Sp34_50242000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>WBGene00000004</td>\n",
       "      <td>Sp34_X0099800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>WBGene00000006</td>\n",
       "      <td>Sp34_10138500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48841</th>\n",
       "      <td>WBGene00305086</td>\n",
       "      <td>Sp34_20266820</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48859</th>\n",
       "      <td>WBGene00305157</td>\n",
       "      <td>Sp34_20128400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48875</th>\n",
       "      <td>WBGene00305999</td>\n",
       "      <td>Sp34_30169600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48877</th>\n",
       "      <td>WBGene00306001</td>\n",
       "      <td>Sp34_30207200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48883</th>\n",
       "      <td>WBGene00306013</td>\n",
       "      <td>Sp34_30302720</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10992 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              elegans      inopinata\n",
       "0      WBGene00000001  Sp34_10257700\n",
       "1      WBGene00000002  Sp34_40340700\n",
       "2      WBGene00000003  Sp34_50242000\n",
       "3      WBGene00000004  Sp34_X0099800\n",
       "5      WBGene00000006  Sp34_10138500\n",
       "...               ...            ...\n",
       "48841  WBGene00305086  Sp34_20266820\n",
       "48859  WBGene00305157  Sp34_20128400\n",
       "48875  WBGene00305999  Sp34_30169600\n",
       "48877  WBGene00306001  Sp34_30207200\n",
       "48883  WBGene00306013  Sp34_30302720\n",
       "\n",
       "[10992 rows x 2 columns]"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orthologous_genes=pd.read_csv('~/part_ii_project/orthologs/elegans_'+worm+'_with_ortholog_type.txt', sep='\\t')\n",
    "\n",
    "one2one=orthologous_genes[orthologous_genes['Homology type']=='ortholog_one2one'].copy()\n",
    "orth_df=one2one.drop('Homology type', axis=1).drop('Genome project', axis=1)\n",
    "orth_df.columns=['elegans',worm]\n",
    "\n",
    "orth_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "resistant-klein",
   "metadata": {},
   "outputs": [],
   "source": [
    "elegans_to_worm={}\n",
    "\n",
    "for index, row in orth_df.iterrows():\n",
    "    elegans_to_worm[row['elegans']]=row[worm]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "unlike-concrete",
   "metadata": {},
   "outputs": [],
   "source": [
    "# WITH OPERONS\n",
    "# this goes through all the genes in the c. elegans gene list and takes only those genes where:\n",
    "# 1) they are present in the list of 1:1 orthologs\n",
    "# 2) their briggsae orthologue is present in the briggsae gene list (some discrepances here due to release #)\n",
    "\n",
    "# it also gathers the indices necessary to reorder the briggsae df so that each ortholog in elegans/briggsae is at the same index\n",
    "\n",
    "elegans_good_genes=[]\n",
    "elegans_idx=[]\n",
    "worm_good_genes=[]\n",
    "worm_idx=[]\n",
    "for index, row in elegans_genes.iterrows():\n",
    "    take=orth_df[orth_df['elegans']==row['name']].copy().reset_index(drop=True)\n",
    "    if len(take)>0:\n",
    "        worm_take=worm_genes[worm_genes['name']==take[worm][0]].copy().reset_index(drop=False)\n",
    "        if len(worm_take)>0:\n",
    "            elegans_good_genes.append(row['name'])\n",
    "            elegans_idx.append(index)\n",
    "            worm_good_genes.append(worm_take['name'][0])\n",
    "            worm_idx.append(worm_take['index'][0])\n",
    "            \n",
    "elegans_final_with_ops=elegans_genes.iloc[elegans_idx,:].copy().reset_index(drop=True)\n",
    "worm_final_with_ops=worm_genes.iloc[worm_idx,:].copy().reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "accurate-escape",
   "metadata": {},
   "outputs": [],
   "source": [
    "# NO OPERONS\n",
    "# this goes through all the genes in the c. elegans gene list and takes only those genes where:\n",
    "# 1) they are present in the list of 1:1 orthologs\n",
    "# 2) their briggsae orthologue is present in the briggsae gene list (some discrepances here due to release #)\n",
    "\n",
    "# it also gathers the indices necessary to reorder the briggsae df so that each ortholog in elegans/briggsae is at the same index\n",
    "\n",
    "elegans_good_genes=[]\n",
    "elegans_idx=[]\n",
    "worm_good_genes=[]\n",
    "worm_idx=[]\n",
    "for index, row in elegans_genes_no_ops.iterrows():\n",
    "    take=orth_df[orth_df['elegans']==row['name']].copy().reset_index(drop=True)\n",
    "    if len(take)>0:\n",
    "        worm_take=worm_genes[worm_genes['name']==take[worm][0]].copy().reset_index(drop=False)\n",
    "        if len(worm_take)>0:\n",
    "            elegans_good_genes.append(row['name'])\n",
    "            elegans_idx.append(index)\n",
    "            worm_good_genes.append(worm_take['name'][0])\n",
    "            worm_idx.append(worm_take['index'][0])\n",
    "            \n",
    "elegans_final_no_ops=elegans_genes_no_ops.iloc[elegans_idx,:].copy().reset_index(drop=True)\n",
    "worm_final_no_ops=worm_genes.iloc[worm_idx,:].copy().reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "consolidated-opposition",
   "metadata": {},
   "outputs": [],
   "source": [
    "worm_final_with_ops.to_csv('~/part_ii_project/canonical_genesets/'+worm+'_to_compare.bed', \n",
    "                     header=None, index=None, sep='\\t')\n",
    "elegans_final_with_ops.to_csv('~/part_ii_project/canonical_genesets/elegans_for_'+worm+'_to_compare.bed',\n",
    "                             header=None, index=None, sep='\\t')\n",
    "\n",
    "worm_final_no_ops.to_csv('~/part_ii_project/canonical_genesets/'+worm+'_orths_no_ops.bed', \n",
    "                     header=None, index=None, sep='\\t')\n",
    "elegans_final_no_ops.to_csv('~/part_ii_project/canonical_genesets/elegans_for_'+worm+'_orths_no_ops.bed',\n",
    "                           header=None, index=None, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "chinese-quilt",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>chromStart</th>\n",
       "      <th>chromEnd</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>thickStart</th>\n",
       "      <th>thickEnd</th>\n",
       "      <th>itemRgb</th>\n",
       "      <th>blockCount</th>\n",
       "      <th>blockSizes</th>\n",
       "      <th>blockStart</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>I</td>\n",
       "      <td>4114</td>\n",
       "      <td>10606</td>\n",
       "      <td>WBGene00022277</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>4220</td>\n",
       "      <td>10148</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>244,102,291,120,136,77,</td>\n",
       "      <td>0,1080,1922,5612,5980,6415,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>I</td>\n",
       "      <td>11494</td>\n",
       "      <td>16837</td>\n",
       "      <td>WBGene00022276</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "      <td>11640</td>\n",
       "      <td>16585</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>67,72,210,365,</td>\n",
       "      <td>0,123,3456,4978,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>I</td>\n",
       "      <td>17482</td>\n",
       "      <td>26781</td>\n",
       "      <td>WBGene00022278</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>17910</td>\n",
       "      <td>26778</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>476,110,227,208,117,127,195,162,200,154,92,49,</td>\n",
       "      <td>0,523,1532,2788,3365,3530,7168,7446,7790,8888,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>I</td>\n",
       "      <td>27588</td>\n",
       "      <td>33150</td>\n",
       "      <td>WBGene00022279</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>28279</td>\n",
       "      <td>32482</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>817,268,656,304,131,114,</td>\n",
       "      <td>0,1511,2180,4180,4823,5448,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>I</td>\n",
       "      <td>93021</td>\n",
       "      <td>94884</td>\n",
       "      <td>WBGene00021676</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "      <td>93091</td>\n",
       "      <td>94714</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>115,375,407,145,300,263,</td>\n",
       "      <td>0,160,587,1042,1247,1600,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8107</th>\n",
       "      <td>X</td>\n",
       "      <td>17662444</td>\n",
       "      <td>17665357</td>\n",
       "      <td>WBGene00003744</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "      <td>17662462</td>\n",
       "      <td>17664159</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>116,146,89,430,</td>\n",
       "      <td>0,876,1629,2483,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8108</th>\n",
       "      <td>X</td>\n",
       "      <td>17671021</td>\n",
       "      <td>17680793</td>\n",
       "      <td>WBGene00020732</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "      <td>17679538</td>\n",
       "      <td>17680396</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>138,86,90,150,469,304,258,459,222,286,72,418,1...</td>\n",
       "      <td>0,184,2369,3096,3337,3924,4343,4742,5249,5751,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8109</th>\n",
       "      <td>X</td>\n",
       "      <td>17680820</td>\n",
       "      <td>17684811</td>\n",
       "      <td>WBGene00044772</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>17680910</td>\n",
       "      <td>17682294</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>205,95,56,</td>\n",
       "      <td>0,1379,3935,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8110</th>\n",
       "      <td>X</td>\n",
       "      <td>17695231</td>\n",
       "      <td>17701744</td>\n",
       "      <td>WBGene00017625</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "      <td>17696474</td>\n",
       "      <td>17701424</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>88,250,236,221,324,132,491,</td>\n",
       "      <td>0,1241,2747,3028,3474,4465,6022,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8111</th>\n",
       "      <td>X</td>\n",
       "      <td>17714967</td>\n",
       "      <td>17718721</td>\n",
       "      <td>WBGene00007068</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "      <td>17714967</td>\n",
       "      <td>17718647</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>144,114,108,116,294,</td>\n",
       "      <td>0,2006,2203,2360,3460,</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8112 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     chrom  chromStart  chromEnd            name  score strand  thickStart  \\\n",
       "0        I        4114     10606  WBGene00022277      0      -        4220   \n",
       "1        I       11494     16837  WBGene00022276      0      +       11640   \n",
       "2        I       17482     26781  WBGene00022278      0      -       17910   \n",
       "3        I       27588     33150  WBGene00022279      0      -       28279   \n",
       "4        I       93021     94884  WBGene00021676      0      +       93091   \n",
       "...    ...         ...       ...             ...    ...    ...         ...   \n",
       "8107     X    17662444  17665357  WBGene00003744      0      +    17662462   \n",
       "8108     X    17671021  17680793  WBGene00020732      0      +    17679538   \n",
       "8109     X    17680820  17684811  WBGene00044772      0      -    17680910   \n",
       "8110     X    17695231  17701744  WBGene00017625      0      +    17696474   \n",
       "8111     X    17714967  17718721  WBGene00007068      0      +    17714967   \n",
       "\n",
       "      thickEnd  itemRgb  blockCount  \\\n",
       "0        10148        0           6   \n",
       "1        16585        0           4   \n",
       "2        26778        0          12   \n",
       "3        32482        0           6   \n",
       "4        94714        0           6   \n",
       "...        ...      ...         ...   \n",
       "8107  17664159        0           4   \n",
       "8108  17680396        0          14   \n",
       "8109  17682294        0           3   \n",
       "8110  17701424        0           7   \n",
       "8111  17718647        0           5   \n",
       "\n",
       "                                             blockSizes  \\\n",
       "0                               244,102,291,120,136,77,   \n",
       "1                                        67,72,210,365,   \n",
       "2        476,110,227,208,117,127,195,162,200,154,92,49,   \n",
       "3                              817,268,656,304,131,114,   \n",
       "4                              115,375,407,145,300,263,   \n",
       "...                                                 ...   \n",
       "8107                                    116,146,89,430,   \n",
       "8108  138,86,90,150,469,304,258,459,222,286,72,418,1...   \n",
       "8109                                         205,95,56,   \n",
       "8110                        88,250,236,221,324,132,491,   \n",
       "8111                               144,114,108,116,294,   \n",
       "\n",
       "                                             blockStart  \n",
       "0                           0,1080,1922,5612,5980,6415,  \n",
       "1                                      0,123,3456,4978,  \n",
       "2     0,523,1532,2788,3365,3530,7168,7446,7790,8888,...  \n",
       "3                           0,1511,2180,4180,4823,5448,  \n",
       "4                             0,160,587,1042,1247,1600,  \n",
       "...                                                 ...  \n",
       "8107                                   0,876,1629,2483,  \n",
       "8108  0,184,2369,3096,3337,3924,4343,4742,5249,5751,...  \n",
       "8109                                       0,1379,3935,  \n",
       "8110                   0,1241,2747,3028,3474,4465,6022,  \n",
       "8111                             0,2006,2203,2360,3460,  \n",
       "\n",
       "[8112 rows x 12 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "elegans_final_no_ops"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "experimental-spectacular",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>chromStart</th>\n",
       "      <th>chromEnd</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>thickStart</th>\n",
       "      <th>thickEnd</th>\n",
       "      <th>itemRgb</th>\n",
       "      <th>blockCount</th>\n",
       "      <th>blockSizes</th>\n",
       "      <th>blockStart</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>I</td>\n",
       "      <td>12509907</td>\n",
       "      <td>12514217</td>\n",
       "      <td>Sp34_10225700</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>12509907</td>\n",
       "      <td>12514217</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>141,102,291,162,</td>\n",
       "      <td>0,295,445,4148,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>I</td>\n",
       "      <td>12514398</td>\n",
       "      <td>12517578</td>\n",
       "      <td>Sp34_10225800</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "      <td>12514398</td>\n",
       "      <td>12517578</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>5,6,62,207,113,</td>\n",
       "      <td>0,1319,1377,2034,3067,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>I</td>\n",
       "      <td>5345749</td>\n",
       "      <td>5348704</td>\n",
       "      <td>Sp34_10093510</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>5345749</td>\n",
       "      <td>5348704</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>78,110,227,208,144,127,195,162,194,139,113,46,</td>\n",
       "      <td>0,127,316,603,888,1241,1473,1725,1936,2306,274...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>I</td>\n",
       "      <td>12519808</td>\n",
       "      <td>12522472</td>\n",
       "      <td>Sp34_10225900</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>12519808</td>\n",
       "      <td>12522472</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>126,256,966,71,</td>\n",
       "      <td>0,181,1228,2593,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>II</td>\n",
       "      <td>18824098</td>\n",
       "      <td>18826602</td>\n",
       "      <td>Sp34_20323200</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>18824098</td>\n",
       "      <td>18826602</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>105,439,407,708,72,</td>\n",
       "      <td>0,154,1171,1631,2432,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8107</th>\n",
       "      <td>X</td>\n",
       "      <td>2218882</td>\n",
       "      <td>2219284</td>\n",
       "      <td>Sp34_X0029400</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>2218882</td>\n",
       "      <td>2219284</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>17,140,98,</td>\n",
       "      <td>0,119,304,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8108</th>\n",
       "      <td>X</td>\n",
       "      <td>2205166</td>\n",
       "      <td>2213498</td>\n",
       "      <td>Sp34_X0029300</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>2205166</td>\n",
       "      <td>2213498</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>112,111,68,57,102,75,286,222,525,243,304,502,9...</td>\n",
       "      <td>0,187,950,1142,1395,2484,2686,3154,3428,4115,4...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8109</th>\n",
       "      <td>X</td>\n",
       "      <td>2203666</td>\n",
       "      <td>2204596</td>\n",
       "      <td>Sp34_X0029200</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "      <td>2203666</td>\n",
       "      <td>2204596</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>36,95,115,</td>\n",
       "      <td>0,531,815,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8110</th>\n",
       "      <td>X</td>\n",
       "      <td>187336</td>\n",
       "      <td>194193</td>\n",
       "      <td>Sp34_X0002300</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>187336</td>\n",
       "      <td>194193</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>234,132,324,221,236,242,</td>\n",
       "      <td>0,321,2777,4986,5730,6615,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8111</th>\n",
       "      <td>X</td>\n",
       "      <td>1950835</td>\n",
       "      <td>1951893</td>\n",
       "      <td>Sp34_X0025900</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>1950835</td>\n",
       "      <td>1951893</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>187,116,108,114,147,</td>\n",
       "      <td>0,386,551,744,911,</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8112 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     chrom  chromStart  chromEnd           name  score strand  thickStart  \\\n",
       "0        I    12509907  12514217  Sp34_10225700      0      -    12509907   \n",
       "1        I    12514398  12517578  Sp34_10225800      0      +    12514398   \n",
       "2        I     5345749   5348704  Sp34_10093510      0      -     5345749   \n",
       "3        I    12519808  12522472  Sp34_10225900      0      -    12519808   \n",
       "4       II    18824098  18826602  Sp34_20323200      0      -    18824098   \n",
       "...    ...         ...       ...            ...    ...    ...         ...   \n",
       "8107     X     2218882   2219284  Sp34_X0029400      0      -     2218882   \n",
       "8108     X     2205166   2213498  Sp34_X0029300      0      -     2205166   \n",
       "8109     X     2203666   2204596  Sp34_X0029200      0      +     2203666   \n",
       "8110     X      187336    194193  Sp34_X0002300      0      -      187336   \n",
       "8111     X     1950835   1951893  Sp34_X0025900      0      -     1950835   \n",
       "\n",
       "      thickEnd  itemRgb  blockCount  \\\n",
       "0     12514217        0           4   \n",
       "1     12517578        0           5   \n",
       "2      5348704        0          12   \n",
       "3     12522472        0           4   \n",
       "4     18826602        0           5   \n",
       "...        ...      ...         ...   \n",
       "8107   2219284        0           3   \n",
       "8108   2213498        0          15   \n",
       "8109   2204596        0           3   \n",
       "8110    194193        0           6   \n",
       "8111   1951893        0           5   \n",
       "\n",
       "                                             blockSizes  \\\n",
       "0                                      141,102,291,162,   \n",
       "1                                       5,6,62,207,113,   \n",
       "2        78,110,227,208,144,127,195,162,194,139,113,46,   \n",
       "3                                       126,256,966,71,   \n",
       "4                                   105,439,407,708,72,   \n",
       "...                                                 ...   \n",
       "8107                                         17,140,98,   \n",
       "8108  112,111,68,57,102,75,286,222,525,243,304,502,9...   \n",
       "8109                                         36,95,115,   \n",
       "8110                           234,132,324,221,236,242,   \n",
       "8111                               187,116,108,114,147,   \n",
       "\n",
       "                                             blockStart  \n",
       "0                                       0,295,445,4148,  \n",
       "1                                0,1319,1377,2034,3067,  \n",
       "2     0,127,316,603,888,1241,1473,1725,1936,2306,274...  \n",
       "3                                      0,181,1228,2593,  \n",
       "4                                 0,154,1171,1631,2432,  \n",
       "...                                                 ...  \n",
       "8107                                         0,119,304,  \n",
       "8108  0,187,950,1142,1395,2484,2686,3154,3428,4115,4...  \n",
       "8109                                         0,531,815,  \n",
       "8110                         0,321,2777,4986,5730,6615,  \n",
       "8111                                 0,386,551,744,911,  \n",
       "\n",
       "[8112 rows x 12 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "worm_final_no_ops"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "creative-collapse",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Sp34_10093510'"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "elegans_to_worm['WBGene00022278']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "lonely-leone",
   "metadata": {},
   "source": [
    "# Get protein sequence per gene"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "later-blowing",
   "metadata": {},
   "source": [
    "'~/part_ii_project/protein_sets/get_protein_sequence_per_gene'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "informative-dating",
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio.Seq import Seq\n",
    "from Bio.SeqRecord import SeqRecord"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "greater-status",
   "metadata": {},
   "outputs": [],
   "source": [
    "path='~/part_ii_project/canonical_genesets/'+worm+'_one_to_one_with_transcript_name.bed'\n",
    "one_to_one=pd.read_csv(path, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "commercial-austin",
   "metadata": {},
   "outputs": [],
   "source": [
    "# may have to remake the 'name column' so that it is in the same format as transcript names in the protein fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "stuck-accommodation",
   "metadata": {},
   "outputs": [],
   "source": [
    "# go through the fasta file downloaded from the ftp site\n",
    "# take only those records with an id that is present in the one_to_one df column with names of longest transcripts\n",
    "\n",
    "to_take=[]\n",
    "\n",
    "path=protein_fasta_file\n",
    "\n",
    "to_take=[record for record in SeqIO.parse(path, 'fasta') if record.id in list(one_to_one['name'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "excess-imperial",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "21443"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(to_take)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "suspended-evanescence",
   "metadata": {},
   "outputs": [],
   "source": [
    "transcript_id_path='/home/helena_hatrick/part_ii_project/protein_sets/'+worm+'_longest_protein_sequences_transcript_id.fasta'\n",
    "SeqIO.write(to_take, transcript_id_path, \"fasta\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "hired-border",
   "metadata": {},
   "outputs": [],
   "source": [
    "t_to_g={}\n",
    "for index, row in one_to_one.iterrows():\n",
    "    t_to_g[row['name']]=row['gene']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "incorporate-television",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Replace transcript ids with gene ids\n",
    "\n",
    "gene_ids=[record for record in SeqIO.parse(transcript_id_path,'fasta')]\n",
    "\n",
    "for record in gene_ids:\n",
    "    record.id=t_to_g[record.id]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "split-recommendation",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "21443"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_id_path='/home/helena_hatrick/part_ii_project/protein_sets/'+worm+'_longest_protein_sequences_gene_id.fasta'\n",
    "SeqIO.write(gene_ids,gene_id_path,\"fasta\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "mental-phone",
   "metadata": {},
   "outputs": [],
   "source": [
    "path='/home/helena_hatrick/part_ii_project/protein_sets/'+worm+'_longest_protein_sequences_gene_id.fasta'\n",
    "    \n",
    "ids=[record.id for record in SeqIO.parse(path,'fasta')]\n",
    "\n",
    "seqs=[record.seq for record in SeqIO.parse(path,'fasta')]\n",
    "\n",
    "initial_descriptions=[record.description for record in SeqIO.parse(path,'fasta')]\n",
    "descriptions=[]\n",
    "for description in initial_descriptions:\n",
    "    split=description.split(' ')[2:]\n",
    "    joined=' '.join([str(x) for x in split])\n",
    "    descriptions.append(joined)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "id": "alternate-faith",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "21443"
      ]
     },
     "execution_count": 106,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "recs=[]\n",
    "for i in range(len(ids)):\n",
    "    \n",
    "    rec=SeqRecord(seqs[i],id=ids[i], description=descriptions[i])\n",
    "    recs.append(rec)\n",
    "\n",
    "path='/home/helena_hatrick/part_ii_project/protein_sets/run_blastp/'+worm+'_gene_id_longest_protein_sequence_with_description.fasta'\n",
    "SeqIO.write(recs,path,'fasta')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "id": "outstanding-scenario",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "21443"
      ]
     },
     "execution_count": 112,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "recs=[]\n",
    "for i in range(len(ids)):\n",
    "    \n",
    "    rec=SeqRecord(seqs[i],id=ids[i])\n",
    "    recs.append(rec)\n",
    "\n",
    "path='/home/helena_hatrick/part_ii_project/protein_sets/run_blastp/'+worm+'_gene_id_longest_protein_sequence_no_description.fasta'\n",
    "SeqIO.write(recs,path,'fasta')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "apart-northern",
   "metadata": {},
   "source": [
    "# Run BLASTp"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "wireless-circuit",
   "metadata": {},
   "source": [
    "'~/part_ii_project/run_blastp'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "continued-chaos",
   "metadata": {},
   "outputs": [],
   "source": [
    "target_worm='elegans'\n",
    "query_worm=worm\n",
    "\n",
    "query_fasta='/home/helena_hatrick/part_ii_project/protein_sets/run_blastp/'+query_worm+'_gene_id_longest_protein_sequence_no_description.fasta'\n",
    "target_fasta='/home/helena_hatrick/part_ii_project/protein_sets/run_blastp/'+target_worm+'_gene_id_longest_protein_sequence_no_description.fasta'\n",
    "target_db='/home/helena_hatrick/part_ii_project/protein_sets/run_blastp/'+target_worm+'_blastdb'\n",
    "\n",
    "! makeblastdb -in {target_fasta} -out {target_db} -dbtype prot\n",
    "\n",
    "out_file='/home/helena_hatrick/part_ii_project/protein_sets/run_blastp/'+target_worm+'_target_vs_'+query_worm+'_query.txt'\n",
    "\n",
    "! blastp -db {target_db} -query {query_fasta} -out {out_file} -outfmt 6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "present-transcription",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "Building a new DB, current time: 03/03/2021 23:37:35\n",
      "New DB name:   /home/helena_hatrick/part_ii_project/protein_sets/run_blastp/inopinata_blastdb\n",
      "New DB title:  /home/helena_hatrick/part_ii_project/protein_sets/run_blastp/inopinata_gene_id_longest_protein_sequence_no_description.fasta\n",
      "Sequence type: Protein\n",
      "Deleted existing Protein BLAST database named /home/helena_hatrick/part_ii_project/protein_sets/run_blastp/inopinata_blastdb\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 21443 sequences in 4.69531 seconds.\n"
     ]
    }
   ],
   "source": [
    "target_worm=worm\n",
    "query_worm='elegans'\n",
    "\n",
    "query_fasta='/home/helena_hatrick/part_ii_project/protein_sets/run_blastp/'+query_worm+'_gene_id_longest_protein_sequence_no_description.fasta'\n",
    "target_fasta='/home/helena_hatrick/part_ii_project/protein_sets/run_blastp/'+target_worm+'_gene_id_longest_protein_sequence_no_description.fasta'\n",
    "target_db='/home/helena_hatrick/part_ii_project/protein_sets/run_blastp/'+target_worm+'_blastdb'\n",
    "\n",
    "! makeblastdb -in {target_fasta} -out {target_db} -dbtype prot\n",
    "\n",
    "out_file='/home/helena_hatrick/part_ii_project/protein_sets/run_blastp/'+target_worm+'_target_vs_'+query_worm+'_query.txt'\n",
    "\n",
    "! blastp -db {target_db} -query {query_fasta} -out {out_file} -outfmt 6"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "inside-grenada",
   "metadata": {},
   "source": [
    "# Make file of RBH ortholog pairs and their e values from BLASTp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "corporate-designation",
   "metadata": {},
   "outputs": [],
   "source": [
    "# read in the blast outputs\n",
    "\n",
    "names=['qseqid','sseqid','pident','length','mismatch','gapopen','qstart','qend','sstart','send','evalue','bitscore']\n",
    "path_stem='/home/helena_hatrick/part_ii_project/protein_sets/run_blastp/'\n",
    "\n",
    "elegans_vs_worm=pd.read_csv(path_stem+'elegans_target_vs_'+worm+'_query.txt',sep='\\t',names=names)\n",
    "\n",
    "worm_vs_elegans=pd.read_csv(path_stem+worm+'_target_vs_elegans_query.txt',sep='\\t',names=names)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "athletic-gathering",
   "metadata": {},
   "outputs": [],
   "source": [
    "operon_options=['with', 'no']\n",
    "which_target=['elegans', worm]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "better-black",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "2000\n",
      "4000\n",
      "6000\n",
      "8000\n",
      "10000\n",
      "12000\n",
      "14000\n",
      "16000\n",
      "18000\n",
      "20000\n",
      "0\n",
      "2000\n",
      "4000\n",
      "6000\n",
      "8000\n",
      "10000\n",
      "12000\n",
      "14000\n",
      "16000\n",
      "18000\n"
     ]
    }
   ],
   "source": [
    "best_hits={}\n",
    "\n",
    "for target in which_target:\n",
    "\n",
    "    if target=='elegans': df=elegans_vs_worm\n",
    "    if target==worm: df=worm_vs_elegans\n",
    "    top_hit_ixs=[]\n",
    "\n",
    "    for num, query_gene in enumerate(list(set(df['qseqid']))):\n",
    "        if num%2000==0:print(num)\n",
    "        subdf=df[df['qseqid']==query_gene].copy()\n",
    "        top_hit_ixs.append(list(subdf.index)[0])\n",
    "\n",
    "    best_hits[target]=df.iloc[top_hit_ixs, :].copy().reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "going-pressure",
   "metadata": {},
   "source": [
    "I will be using the dataset with elegans as target for all future things, not that it should matter."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "bound-grant",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>qseqid</th>\n",
       "      <th>sseqid</th>\n",
       "      <th>pident</th>\n",
       "      <th>length</th>\n",
       "      <th>mismatch</th>\n",
       "      <th>gapopen</th>\n",
       "      <th>qstart</th>\n",
       "      <th>qend</th>\n",
       "      <th>sstart</th>\n",
       "      <th>send</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Sp34_50429700</td>\n",
       "      <td>WBGene00044183</td>\n",
       "      <td>39.474</td>\n",
       "      <td>38</td>\n",
       "      <td>16</td>\n",
       "      <td>1</td>\n",
       "      <td>81</td>\n",
       "      <td>111</td>\n",
       "      <td>2</td>\n",
       "      <td>39</td>\n",
       "      <td>2.000000e+00</td>\n",
       "      <td>25.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Sp34_30105200</td>\n",
       "      <td>WBGene00011734</td>\n",
       "      <td>97.183</td>\n",
       "      <td>142</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>142</td>\n",
       "      <td>1</td>\n",
       "      <td>142</td>\n",
       "      <td>6.170000e-101</td>\n",
       "      <td>284.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Sp34_20183600</td>\n",
       "      <td>WBGene00010711</td>\n",
       "      <td>27.389</td>\n",
       "      <td>157</td>\n",
       "      <td>95</td>\n",
       "      <td>8</td>\n",
       "      <td>137</td>\n",
       "      <td>290</td>\n",
       "      <td>1</td>\n",
       "      <td>141</td>\n",
       "      <td>5.810000e-07</td>\n",
       "      <td>48.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Sp34_10273500</td>\n",
       "      <td>WBGene00007700</td>\n",
       "      <td>74.474</td>\n",
       "      <td>380</td>\n",
       "      <td>82</td>\n",
       "      <td>4</td>\n",
       "      <td>53</td>\n",
       "      <td>432</td>\n",
       "      <td>18</td>\n",
       "      <td>382</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>577.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Sp34_50230810</td>\n",
       "      <td>WBGene00019203</td>\n",
       "      <td>87.662</td>\n",
       "      <td>154</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>154</td>\n",
       "      <td>1</td>\n",
       "      <td>154</td>\n",
       "      <td>1.190000e-102</td>\n",
       "      <td>290.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21417</th>\n",
       "      <td>Sp34_50402510</td>\n",
       "      <td>WBGene00012615</td>\n",
       "      <td>75.556</td>\n",
       "      <td>90</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>90</td>\n",
       "      <td>1</td>\n",
       "      <td>90</td>\n",
       "      <td>2.270000e-41</td>\n",
       "      <td>130.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21418</th>\n",
       "      <td>Sp34_20217500</td>\n",
       "      <td>WBGene00004038</td>\n",
       "      <td>91.790</td>\n",
       "      <td>1352</td>\n",
       "      <td>105</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>1348</td>\n",
       "      <td>1</td>\n",
       "      <td>1350</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2574.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21419</th>\n",
       "      <td>Sp34_X0160500</td>\n",
       "      <td>WBGene00001425</td>\n",
       "      <td>85.430</td>\n",
       "      <td>151</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>151</td>\n",
       "      <td>1</td>\n",
       "      <td>151</td>\n",
       "      <td>2.520000e-82</td>\n",
       "      <td>238.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21420</th>\n",
       "      <td>Sp34_20091000</td>\n",
       "      <td>WBGene00011464</td>\n",
       "      <td>23.784</td>\n",
       "      <td>185</td>\n",
       "      <td>121</td>\n",
       "      <td>7</td>\n",
       "      <td>27</td>\n",
       "      <td>197</td>\n",
       "      <td>47</td>\n",
       "      <td>225</td>\n",
       "      <td>6.000000e-03</td>\n",
       "      <td>37.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21421</th>\n",
       "      <td>Sp34_X0187000</td>\n",
       "      <td>WBGene00003053</td>\n",
       "      <td>69.198</td>\n",
       "      <td>237</td>\n",
       "      <td>71</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>235</td>\n",
       "      <td>1</td>\n",
       "      <td>237</td>\n",
       "      <td>2.100000e-115</td>\n",
       "      <td>328.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>21422 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              qseqid          sseqid  pident  length  mismatch  gapopen  \\\n",
       "0      Sp34_50429700  WBGene00044183  39.474      38        16        1   \n",
       "1      Sp34_30105200  WBGene00011734  97.183     142         4        0   \n",
       "2      Sp34_20183600  WBGene00010711  27.389     157        95        8   \n",
       "3      Sp34_10273500  WBGene00007700  74.474     380        82        4   \n",
       "4      Sp34_50230810  WBGene00019203  87.662     154        19        0   \n",
       "...              ...             ...     ...     ...       ...      ...   \n",
       "21417  Sp34_50402510  WBGene00012615  75.556      90        22        0   \n",
       "21418  Sp34_20217500  WBGene00004038  91.790    1352       105        4   \n",
       "21419  Sp34_X0160500  WBGene00001425  85.430     151        22        0   \n",
       "21420  Sp34_20091000  WBGene00011464  23.784     185       121        7   \n",
       "21421  Sp34_X0187000  WBGene00003053  69.198     237        71        1   \n",
       "\n",
       "       qstart  qend  sstart  send         evalue  bitscore  \n",
       "0          81   111       2    39   2.000000e+00      25.8  \n",
       "1           1   142       1   142  6.170000e-101     284.0  \n",
       "2         137   290       1   141   5.810000e-07      48.1  \n",
       "3          53   432      18   382   0.000000e+00     577.0  \n",
       "4           1   154       1   154  1.190000e-102     290.0  \n",
       "...       ...   ...     ...   ...            ...       ...  \n",
       "21417       1    90       1    90   2.270000e-41     130.0  \n",
       "21418       1  1348       1  1350   0.000000e+00    2574.0  \n",
       "21419       1   151       1   151   2.520000e-82     238.0  \n",
       "21420      27   197      47   225   6.000000e-03      37.0  \n",
       "21421       1   235       1   237  2.100000e-115     328.0  \n",
       "\n",
       "[21422 rows x 12 columns]"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "best_hits['elegans']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "straight-rotation",
   "metadata": {},
   "outputs": [],
   "source": [
    "rbh_indices=[]\n",
    "\n",
    "e_target=best_hits['elegans']\n",
    "x_target=best_hits[worm]\n",
    "\n",
    "for index, row in e_target.iterrows():\n",
    "\n",
    "    hit = row['sseqid']\n",
    "    test=x_target[x_target['qseqid']==hit].reset_index(drop=True)\n",
    "    if test['sseqid'][0]==row['qseqid']:rbh_indices.append(index)\n",
    "\n",
    "rbhs=e_target.iloc[rbh_indices, :].copy().reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "experienced-general",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>qseqid</th>\n",
       "      <th>sseqid</th>\n",
       "      <th>pident</th>\n",
       "      <th>length</th>\n",
       "      <th>mismatch</th>\n",
       "      <th>gapopen</th>\n",
       "      <th>qstart</th>\n",
       "      <th>qend</th>\n",
       "      <th>sstart</th>\n",
       "      <th>send</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Sp34_50429700</td>\n",
       "      <td>WBGene00044183</td>\n",
       "      <td>39.474</td>\n",
       "      <td>38</td>\n",
       "      <td>16</td>\n",
       "      <td>1</td>\n",
       "      <td>81</td>\n",
       "      <td>111</td>\n",
       "      <td>2</td>\n",
       "      <td>39</td>\n",
       "      <td>2.000000e+00</td>\n",
       "      <td>25.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Sp34_30105200</td>\n",
       "      <td>WBGene00011734</td>\n",
       "      <td>97.183</td>\n",
       "      <td>142</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>142</td>\n",
       "      <td>1</td>\n",
       "      <td>142</td>\n",
       "      <td>6.170000e-101</td>\n",
       "      <td>284.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Sp34_10273500</td>\n",
       "      <td>WBGene00007700</td>\n",
       "      <td>74.474</td>\n",
       "      <td>380</td>\n",
       "      <td>82</td>\n",
       "      <td>4</td>\n",
       "      <td>53</td>\n",
       "      <td>432</td>\n",
       "      <td>18</td>\n",
       "      <td>382</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>577.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Sp34_50230810</td>\n",
       "      <td>WBGene00019203</td>\n",
       "      <td>87.662</td>\n",
       "      <td>154</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>154</td>\n",
       "      <td>1</td>\n",
       "      <td>154</td>\n",
       "      <td>1.190000e-102</td>\n",
       "      <td>290.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Sp34_20044610</td>\n",
       "      <td>WBGene00022448</td>\n",
       "      <td>90.062</td>\n",
       "      <td>161</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>161</td>\n",
       "      <td>1</td>\n",
       "      <td>161</td>\n",
       "      <td>1.400000e-110</td>\n",
       "      <td>310.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12639</th>\n",
       "      <td>Sp34_50277900</td>\n",
       "      <td>WBGene00011284</td>\n",
       "      <td>86.014</td>\n",
       "      <td>143</td>\n",
       "      <td>20</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>143</td>\n",
       "      <td>1</td>\n",
       "      <td>143</td>\n",
       "      <td>2.470000e-91</td>\n",
       "      <td>260.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12640</th>\n",
       "      <td>Sp34_50402510</td>\n",
       "      <td>WBGene00012615</td>\n",
       "      <td>75.556</td>\n",
       "      <td>90</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>90</td>\n",
       "      <td>1</td>\n",
       "      <td>90</td>\n",
       "      <td>2.270000e-41</td>\n",
       "      <td>130.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12641</th>\n",
       "      <td>Sp34_20217500</td>\n",
       "      <td>WBGene00004038</td>\n",
       "      <td>91.790</td>\n",
       "      <td>1352</td>\n",
       "      <td>105</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>1348</td>\n",
       "      <td>1</td>\n",
       "      <td>1350</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2574.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12642</th>\n",
       "      <td>Sp34_X0160500</td>\n",
       "      <td>WBGene00001425</td>\n",
       "      <td>85.430</td>\n",
       "      <td>151</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>151</td>\n",
       "      <td>1</td>\n",
       "      <td>151</td>\n",
       "      <td>2.520000e-82</td>\n",
       "      <td>238.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12643</th>\n",
       "      <td>Sp34_X0187000</td>\n",
       "      <td>WBGene00003053</td>\n",
       "      <td>69.198</td>\n",
       "      <td>237</td>\n",
       "      <td>71</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>235</td>\n",
       "      <td>1</td>\n",
       "      <td>237</td>\n",
       "      <td>2.100000e-115</td>\n",
       "      <td>328.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12644 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              qseqid          sseqid  pident  length  mismatch  gapopen  \\\n",
       "0      Sp34_50429700  WBGene00044183  39.474      38        16        1   \n",
       "1      Sp34_30105200  WBGene00011734  97.183     142         4        0   \n",
       "2      Sp34_10273500  WBGene00007700  74.474     380        82        4   \n",
       "3      Sp34_50230810  WBGene00019203  87.662     154        19        0   \n",
       "4      Sp34_20044610  WBGene00022448  90.062     161        16        0   \n",
       "...              ...             ...     ...     ...       ...      ...   \n",
       "12639  Sp34_50277900  WBGene00011284  86.014     143        20        0   \n",
       "12640  Sp34_50402510  WBGene00012615  75.556      90        22        0   \n",
       "12641  Sp34_20217500  WBGene00004038  91.790    1352       105        4   \n",
       "12642  Sp34_X0160500  WBGene00001425  85.430     151        22        0   \n",
       "12643  Sp34_X0187000  WBGene00003053  69.198     237        71        1   \n",
       "\n",
       "       qstart  qend  sstart  send         evalue  bitscore  \n",
       "0          81   111       2    39   2.000000e+00      25.8  \n",
       "1           1   142       1   142  6.170000e-101     284.0  \n",
       "2          53   432      18   382   0.000000e+00     577.0  \n",
       "3           1   154       1   154  1.190000e-102     290.0  \n",
       "4           1   161       1   161  1.400000e-110     310.0  \n",
       "...       ...   ...     ...   ...            ...       ...  \n",
       "12639       1   143       1   143   2.470000e-91     260.0  \n",
       "12640       1    90       1    90   2.270000e-41     130.0  \n",
       "12641       1  1348       1  1350   0.000000e+00    2574.0  \n",
       "12642       1   151       1   151   2.520000e-82     238.0  \n",
       "12643       1   235       1   237  2.100000e-115     328.0  \n",
       "\n",
       "[12644 rows x 12 columns]"
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rbhs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "traditional-criminal",
   "metadata": {},
   "outputs": [],
   "source": [
    "worm_final_with_ops=pd.read_csv('~/part_ii_project/canonical_genesets/'+worm+'_to_compare.bed', \n",
    "                     names=bed_12_columns, sep='\\t')\n",
    "elegans_final_with_ops=pd.read_csv('~/part_ii_project/canonical_genesets/elegans_for_'+worm+'_to_compare.bed',\n",
    "                             names=bed_12_columns, sep='\\t')\n",
    "\n",
    "worm_final_no_ops=pd.read_csv('~/part_ii_project/canonical_genesets/'+worm+'_orths_no_ops.bed', \n",
    "                     names=bed_12_columns, sep='\\t')\n",
    "elegans_final_no_ops=pd.read_csv('~/part_ii_project/canonical_genesets/elegans_for_'+worm+'_orths_no_ops.bed',\n",
    "                           names=bed_12_columns, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "classical-grove",
   "metadata": {},
   "outputs": [],
   "source": [
    "ortholog_rbhs={}\n",
    "\n",
    "for is_ops in operon_options:\n",
    "    \n",
    "    df=rbhs.copy()\n",
    "    ortholog_rbh_indices=[]\n",
    "    if is_ops=='with': ortholog_df=elegans_final_with_ops\n",
    "    if is_ops=='no':ortholog_df=elegans_final_no_ops\n",
    "    \n",
    "    for index, row in df.iterrows():\n",
    "        if row['sseqid'] in set(ortholog_df['name']):\n",
    "            if elegans_to_worm[row['sseqid']]==row['qseqid']: ortholog_rbh_indices.append(index)\n",
    "            \n",
    "    ortholog_rbhs[is_ops]=df.iloc[ortholog_rbh_indices,:].copy().reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "stone-jacob",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>qseqid</th>\n",
       "      <th>sseqid</th>\n",
       "      <th>pident</th>\n",
       "      <th>length</th>\n",
       "      <th>mismatch</th>\n",
       "      <th>gapopen</th>\n",
       "      <th>qstart</th>\n",
       "      <th>qend</th>\n",
       "      <th>sstart</th>\n",
       "      <th>send</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Sp34_50230810</td>\n",
       "      <td>WBGene00019203</td>\n",
       "      <td>87.662</td>\n",
       "      <td>154</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>154</td>\n",
       "      <td>1</td>\n",
       "      <td>154</td>\n",
       "      <td>1.190000e-102</td>\n",
       "      <td>290.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Sp34_20153100</td>\n",
       "      <td>WBGene00008052</td>\n",
       "      <td>74.010</td>\n",
       "      <td>404</td>\n",
       "      <td>105</td>\n",
       "      <td>0</td>\n",
       "      <td>45</td>\n",
       "      <td>448</td>\n",
       "      <td>1</td>\n",
       "      <td>404</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>639.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Sp34_50092700</td>\n",
       "      <td>WBGene00021493</td>\n",
       "      <td>83.491</td>\n",
       "      <td>424</td>\n",
       "      <td>69</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>433</td>\n",
       "      <td>10</td>\n",
       "      <td>433</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>751.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Sp34_X0144500</td>\n",
       "      <td>WBGene00008094</td>\n",
       "      <td>86.604</td>\n",
       "      <td>739</td>\n",
       "      <td>98</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>739</td>\n",
       "      <td>1</td>\n",
       "      <td>738</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1335.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Sp34_20032600</td>\n",
       "      <td>WBGene00012602</td>\n",
       "      <td>83.824</td>\n",
       "      <td>68</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>68</td>\n",
       "      <td>1</td>\n",
       "      <td>68</td>\n",
       "      <td>6.790000e-41</td>\n",
       "      <td>127.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7910</th>\n",
       "      <td>Sp34_X0248300</td>\n",
       "      <td>WBGene00006475</td>\n",
       "      <td>93.262</td>\n",
       "      <td>564</td>\n",
       "      <td>38</td>\n",
       "      <td>0</td>\n",
       "      <td>43</td>\n",
       "      <td>606</td>\n",
       "      <td>1</td>\n",
       "      <td>564</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1073.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7911</th>\n",
       "      <td>Sp34_50402510</td>\n",
       "      <td>WBGene00012615</td>\n",
       "      <td>75.556</td>\n",
       "      <td>90</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>90</td>\n",
       "      <td>1</td>\n",
       "      <td>90</td>\n",
       "      <td>2.270000e-41</td>\n",
       "      <td>130.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7912</th>\n",
       "      <td>Sp34_20217500</td>\n",
       "      <td>WBGene00004038</td>\n",
       "      <td>91.790</td>\n",
       "      <td>1352</td>\n",
       "      <td>105</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>1348</td>\n",
       "      <td>1</td>\n",
       "      <td>1350</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2574.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7913</th>\n",
       "      <td>Sp34_X0160500</td>\n",
       "      <td>WBGene00001425</td>\n",
       "      <td>85.430</td>\n",
       "      <td>151</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>151</td>\n",
       "      <td>1</td>\n",
       "      <td>151</td>\n",
       "      <td>2.520000e-82</td>\n",
       "      <td>238.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7914</th>\n",
       "      <td>Sp34_X0187000</td>\n",
       "      <td>WBGene00003053</td>\n",
       "      <td>69.198</td>\n",
       "      <td>237</td>\n",
       "      <td>71</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>235</td>\n",
       "      <td>1</td>\n",
       "      <td>237</td>\n",
       "      <td>2.100000e-115</td>\n",
       "      <td>328.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7915 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             qseqid          sseqid  pident  length  mismatch  gapopen  \\\n",
       "0     Sp34_50230810  WBGene00019203  87.662     154        19        0   \n",
       "1     Sp34_20153100  WBGene00008052  74.010     404       105        0   \n",
       "2     Sp34_50092700  WBGene00021493  83.491     424        69        1   \n",
       "3     Sp34_X0144500  WBGene00008094  86.604     739        98        1   \n",
       "4     Sp34_20032600  WBGene00012602  83.824      68        11        0   \n",
       "...             ...             ...     ...     ...       ...      ...   \n",
       "7910  Sp34_X0248300  WBGene00006475  93.262     564        38        0   \n",
       "7911  Sp34_50402510  WBGene00012615  75.556      90        22        0   \n",
       "7912  Sp34_20217500  WBGene00004038  91.790    1352       105        4   \n",
       "7913  Sp34_X0160500  WBGene00001425  85.430     151        22        0   \n",
       "7914  Sp34_X0187000  WBGene00003053  69.198     237        71        1   \n",
       "\n",
       "      qstart  qend  sstart  send         evalue  bitscore  \n",
       "0          1   154       1   154  1.190000e-102     290.0  \n",
       "1         45   448       1   404   0.000000e+00     639.0  \n",
       "2         11   433      10   433   0.000000e+00     751.0  \n",
       "3          1   739       1   738   0.000000e+00    1335.0  \n",
       "4          1    68       1    68   6.790000e-41     127.0  \n",
       "...      ...   ...     ...   ...            ...       ...  \n",
       "7910      43   606       1   564   0.000000e+00    1073.0  \n",
       "7911       1    90       1    90   2.270000e-41     130.0  \n",
       "7912       1  1348       1  1350   0.000000e+00    2574.0  \n",
       "7913       1   151       1   151   2.520000e-82     238.0  \n",
       "7914       1   235       1   237  2.100000e-115     328.0  \n",
       "\n",
       "[7915 rows x 12 columns]"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ortholog_rbhs['no']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "id": "found-negotiation",
   "metadata": {},
   "outputs": [],
   "source": [
    "ortholog_rbhs_neat={}\n",
    "\n",
    "for is_ops in operon_options:\n",
    "    \n",
    "    df=ortholog_rbhs[is_ops]\n",
    "    neat_df=pd.DataFrame({'query_gene':df['qseqid'], 'target_gene':df['sseqid'], 'e_value':df['evalue']})\n",
    "    path='~/part_ii_project/orthologs/'+worm+'_query_121_ortholog_rbhs_'+is_ops+'_ops'\n",
    "    neat_df.to_csv(path, header=['query','target','e_value'], index=None)\n",
    "    ortholog_rbhs_neat[is_ops]=neat_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "id": "interim-culture",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>query_gene</th>\n",
       "      <th>target_gene</th>\n",
       "      <th>e_value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Sp34_30105200</td>\n",
       "      <td>WBGene00011734</td>\n",
       "      <td>6.170000e-101</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Sp34_50230810</td>\n",
       "      <td>WBGene00019203</td>\n",
       "      <td>1.190000e-102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Sp34_20044610</td>\n",
       "      <td>WBGene00022448</td>\n",
       "      <td>1.400000e-110</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Sp34_20153100</td>\n",
       "      <td>WBGene00008052</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Sp34_50092700</td>\n",
       "      <td>WBGene00021493</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10755</th>\n",
       "      <td>Sp34_50277900</td>\n",
       "      <td>WBGene00011284</td>\n",
       "      <td>2.470000e-91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10756</th>\n",
       "      <td>Sp34_50402510</td>\n",
       "      <td>WBGene00012615</td>\n",
       "      <td>2.270000e-41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10757</th>\n",
       "      <td>Sp34_20217500</td>\n",
       "      <td>WBGene00004038</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10758</th>\n",
       "      <td>Sp34_X0160500</td>\n",
       "      <td>WBGene00001425</td>\n",
       "      <td>2.520000e-82</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10759</th>\n",
       "      <td>Sp34_X0187000</td>\n",
       "      <td>WBGene00003053</td>\n",
       "      <td>2.100000e-115</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10760 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          query_gene     target_gene        e_value\n",
       "0      Sp34_30105200  WBGene00011734  6.170000e-101\n",
       "1      Sp34_50230810  WBGene00019203  1.190000e-102\n",
       "2      Sp34_20044610  WBGene00022448  1.400000e-110\n",
       "3      Sp34_20153100  WBGene00008052   0.000000e+00\n",
       "4      Sp34_50092700  WBGene00021493   0.000000e+00\n",
       "...              ...             ...            ...\n",
       "10755  Sp34_50277900  WBGene00011284   2.470000e-91\n",
       "10756  Sp34_50402510  WBGene00012615   2.270000e-41\n",
       "10757  Sp34_20217500  WBGene00004038   0.000000e+00\n",
       "10758  Sp34_X0160500  WBGene00001425   2.520000e-82\n",
       "10759  Sp34_X0187000  WBGene00003053  2.100000e-115\n",
       "\n",
       "[10760 rows x 3 columns]"
      ]
     },
     "execution_count": 103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ortholog_rbhs_neat['with']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "apparent-intake",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>query</th>\n",
       "      <th>target</th>\n",
       "      <th>e_value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Sp34_30105200</td>\n",
       "      <td>WBGene00011734</td>\n",
       "      <td>6.170000e-101</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Sp34_50230810</td>\n",
       "      <td>WBGene00019203</td>\n",
       "      <td>1.190000e-102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Sp34_20044610</td>\n",
       "      <td>WBGene00022448</td>\n",
       "      <td>1.400000e-110</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Sp34_20153100</td>\n",
       "      <td>WBGene00008052</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Sp34_50092700</td>\n",
       "      <td>WBGene00021493</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10755</th>\n",
       "      <td>Sp34_50277900</td>\n",
       "      <td>WBGene00011284</td>\n",
       "      <td>2.470000e-91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10756</th>\n",
       "      <td>Sp34_50402510</td>\n",
       "      <td>WBGene00012615</td>\n",
       "      <td>2.270000e-41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10757</th>\n",
       "      <td>Sp34_20217500</td>\n",
       "      <td>WBGene00004038</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10758</th>\n",
       "      <td>Sp34_X0160500</td>\n",
       "      <td>WBGene00001425</td>\n",
       "      <td>2.520000e-82</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10759</th>\n",
       "      <td>Sp34_X0187000</td>\n",
       "      <td>WBGene00003053</td>\n",
       "      <td>2.100000e-115</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10760 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               query          target        e_value\n",
       "0      Sp34_30105200  WBGene00011734  6.170000e-101\n",
       "1      Sp34_50230810  WBGene00019203  1.190000e-102\n",
       "2      Sp34_20044610  WBGene00022448  1.400000e-110\n",
       "3      Sp34_20153100  WBGene00008052   0.000000e+00\n",
       "4      Sp34_50092700  WBGene00021493   0.000000e+00\n",
       "...              ...             ...            ...\n",
       "10755  Sp34_50277900  WBGene00011284   2.470000e-91\n",
       "10756  Sp34_50402510  WBGene00012615   2.270000e-41\n",
       "10757  Sp34_20217500  WBGene00004038   0.000000e+00\n",
       "10758  Sp34_X0160500  WBGene00001425   2.520000e-82\n",
       "10759  Sp34_X0187000  WBGene00003053  2.100000e-115\n",
       "\n",
       "[10760 rows x 3 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "worm='inopinata'\n",
    "is_ops='with'\n",
    "test=pd.read_csv('~/part_ii_project/orthologs/'+worm+'_query_121_ortholog_rbhs_'+is_ops+'_ops')\n",
    "test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "catholic-crime",
   "metadata": {},
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: '/home/helena_hatrick/part_ii_project/orthologs/briggsae_query_121_ortholog_rbhs_with_ops'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-8-9dae42208e73>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0mworm\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'briggsae'\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[0mis_ops\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'with'\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 3\u001b[0;31m \u001b[0mtest\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread_csv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'~/part_ii_project/orthologs/'\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0mworm\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m'_query_121_ortholog_rbhs_'\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0mis_ops\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m'_ops'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      4\u001b[0m \u001b[0mtest\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/part_ii_project/lib/python3.7/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36mread_csv\u001b[0;34m(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, squeeze, prefix, mangle_dupe_cols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, dialect, error_bad_lines, warn_bad_lines, delim_whitespace, low_memory, memory_map, float_precision, storage_options)\u001b[0m\n\u001b[1;32m    603\u001b[0m     \u001b[0mkwds\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mupdate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkwds_defaults\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    604\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 605\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0m_read\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilepath_or_buffer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    606\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    607\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/part_ii_project/lib/python3.7/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m_read\u001b[0;34m(filepath_or_buffer, kwds)\u001b[0m\n\u001b[1;32m    455\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    456\u001b[0m     \u001b[0;31m# Create the parser.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 457\u001b[0;31m     \u001b[0mparser\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mTextFileReader\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilepath_or_buffer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    458\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    459\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mchunksize\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0miterator\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/part_ii_project/lib/python3.7/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, f, engine, **kwds)\u001b[0m\n\u001b[1;32m    812\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0moptions\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"has_index_names\"\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mkwds\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"has_index_names\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    813\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 814\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_make_engine\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mengine\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    815\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    816\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mclose\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/part_ii_project/lib/python3.7/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m_make_engine\u001b[0;34m(self, engine)\u001b[0m\n\u001b[1;32m   1043\u001b[0m             )\n\u001b[1;32m   1044\u001b[0m         \u001b[0;31m# error: Too many arguments for \"ParserBase\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1045\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mmapping\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mengine\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0moptions\u001b[0m\u001b[0;34m)\u001b[0m  \u001b[0;31m# type: ignore[call-arg]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1046\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1047\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_failover_to_python\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/part_ii_project/lib/python3.7/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, src, **kwds)\u001b[0m\n\u001b[1;32m   1860\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1861\u001b[0m         \u001b[0;31m# open handles\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1862\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_open_handles\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msrc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1863\u001b[0m         \u001b[0;32massert\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhandles\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1864\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0mkey\u001b[0m \u001b[0;32min\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0;34m\"storage_options\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"encoding\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"memory_map\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"compression\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/part_ii_project/lib/python3.7/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m_open_handles\u001b[0;34m(self, src, kwds)\u001b[0m\n\u001b[1;32m   1361\u001b[0m             \u001b[0mcompression\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mkwds\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"compression\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1362\u001b[0m             \u001b[0mmemory_map\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mkwds\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"memory_map\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1363\u001b[0;31m             \u001b[0mstorage_options\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mkwds\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"storage_options\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1364\u001b[0m         )\n\u001b[1;32m   1365\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/envs/part_ii_project/lib/python3.7/site-packages/pandas/io/common.py\u001b[0m in \u001b[0;36mget_handle\u001b[0;34m(path_or_buf, mode, encoding, compression, memory_map, is_text, errors, storage_options)\u001b[0m\n\u001b[1;32m    642\u001b[0m                 \u001b[0mencoding\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mioargs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mencoding\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    643\u001b[0m                 \u001b[0merrors\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0merrors\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 644\u001b[0;31m                 \u001b[0mnewline\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    645\u001b[0m             )\n\u001b[1;32m    646\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: '/home/helena_hatrick/part_ii_project/orthologs/briggsae_query_121_ortholog_rbhs_with_ops'"
     ]
    }
   ],
   "source": [
    "worm='briggsae'\n",
    "is_ops='with'\n",
    "test=pd.read_csv('~/part_ii_project/orthologs/'+worm+'_query_121_ortholog_rbhs_'+is_ops+'_ops')\n",
    "test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "swedish-marker",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "instructional-bailey",
   "metadata": {},
   "source": [
    "'~/part_ii_project/get_evalues_no_ops'"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "amazing-comfort",
   "metadata": {},
   "source": [
    "'~/part_ii_project/get_evalues_with_ops'"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
